File: api/bloom.py
from typing import Dict
import random
import hashlib
import base64
import database_client as db
import settings

def create_bloom_app_from_chat_message(message: str, db_client, pipeline: str = settings.OLD_PIPELINE):
    """
    Simulates creating a BloomApp from a chat message and pipeline.
    Returns a BloomApp dict with a unique id, an image from the specified pipeline, and the origin_pipeline.
    """
    # Get all bloom apps from storage
    all_apps = db_client.read("bloom_apps")
    
    if not all_apps:
        raise RuntimeError("No bloom apps found in database")
    
    # Filter apps by the specified pipeline
    pipeline_apps = {}
    for app_id, app_data in all_apps.items():
        if app_data.get("origin_pipeline") == pipeline:
            pipeline_apps[app_id] = app_data
    
    if pipeline_apps:
        # Get a random app from this pipeline
        app = random.choice(list(pipeline_apps.values()))
        return [app]
    else:
        # Fallback: raise an error if no apps exist for this pipeline
        raise RuntimeError(f"No apps found for pipeline: {pipeline}")

File: api/bloom.ts
import * as fs from 'fs';
import * as path from 'path';
import { v4 as uuidv4 } from 'uuid';
import { BloomApp } from './datatypes';
import { addApp } from './database-client';
import { OLD_PIPELINE } from './settings';

export function createBloomAppFromChatMessage(
  message: string, 
  dbClient: any, 
  pipeline: string = OLD_PIPELINE
): BloomApp[] {
  /**
   * Simulates creating a BloomApp from a chat message and pipeline.
   * Picks a random image from the pipeline folder, assigns a UUID, and stores it in the in-memory database.
   * Returns a base64 data URL for the image (not a file path).
   */
  
  // Find the pipeline folder - back to same directory level
  const resourceDir = path.join(__dirname, "resroucre", pipeline);
  const items = fs.readdirSync(resourceDir);
  const images = items.filter(f => f.endsWith('.png'));
  
  if (images.length === 0) {
    throw new Error(`No images found for pipeline: ${pipeline}`);
  }
  
  const chosenImage = images[Math.floor(Math.random() * images.length)];
  const imagePath = path.join(resourceDir, chosenImage);

  // Read and encode the image as base64 data URL
  const imageBytes = fs.readFileSync(imagePath);
  const encodedImage = imageBytes.toString('base64');
  const imageDataUrl = `data:image/png;base64,${encodedImage}`;

  // Generate a UUID for the app/image ID
  const appId = uuidv4();

  // Store in the in-memory database
  const app: BloomApp = {
    id: appId,
    image: imageDataUrl, // Return the data URL, not the file path
    origin_pipeline: pipeline,
    message: message
  };
  
  addApp(app);

  // Return the new app in a list (to match previous return type)
  return [app];
} 